### DISCUSSION TITLE
How should DataTable headers be aligned for numerical/currency data?

### DISCUSSION THREAD

**Discussion by grahamharper** [carbon-design-system/carbon#14777] [Help]
How should DataTable column headers be aligned when the column contains a numerical/currency value (assuming the data follows formatting best practice by being set to align-right)?

**As-Is example:**
![Screenshot 2023-10-04 at 15 30 59](https://github.com/carbon-design-system/carbon/assets/2853295/42fcb538-d57e-4612-9445-c728923ac61c)
![Screenshot 2023-10-04 at 15 31 16](https://github.com/carbon-design-system/carbon/assets/2853295/6939f8bb-dfbf-48cd-bfaa-e64feac02c57)

I would have expected (and a quick web search confirms) that the column header should be aligned with the data, so in the case of numerical/currency values, the column header should be right aligned.

E.g.
![image](https://github.com/carbon-design-system/carbon/assets/2853295/e5f6366f-d9b3-4ac5-93aa-5966ef0f62c2)

If the Carbon team agrees with this, then should I open a github issue requesting support for right-alignment in column headers, and specifically _sortable_ column headers.

**Comment by thyhmdo** [comment]
@grahamharper that follows the rtl logic that we offer through Text direction. Just need to confirm with a dev if we already support this. @tw15egan @tay1orjones 

<img width="665" alt="image" src="https://github.com/carbon-design-system/carbon/assets/77697351/0647c187-5b1b-4ca9-8bfd-60dd6ead5cbf">

**Reply by grahamharper** [reply]
Thanks. For RTL the reversing of the order of the icon and heading label is automatic based on the text direction mode with flexbox being used here.

What we need is the ability to specify certain columns to align in this manner. In theory I guess we could set `dir='rtl'` on the column header but that doesn't seem correct, and more of a hack.

**Reply by tw15egan** [reply]
Yeah, I think this makes sense because even in RTL languages I can imagine they may want left-aligned currency. Could see an `isCurrencyColumn` prop, or maybe something more agnostic? Are there other use cases besides currency (that we would want to allow?)

**Reply by thyhmdo** [reply]
I see. So that could only applies to a certain column. Can this prop be more general so people can turn it on if they want? I'm not sure what others use cases are to have this different direction.

**Reply by grahamharper** [reply]
Something like `isNumeric` on the column might better as it's not just currency values, could be  numbers representing a count of something, like population size in table of countries.

> Alignment Matters
> 3½ simple rules to follow:
> 1. Numerical data is right-aligned
> 2. Textual data is left-aligned
> 3. Headers are aligned with their data
> 3½. Don’t use center alignment.

> Numerical data is read right-to-left; that is, we compare numbers by first looking at their ones digit, then their tens, then their hundreds, and so on. This is also how most people learn arithmetic — start on the right and move left, carrying digits as you go[1]. Therefore, tables should keep numerical data right-aligned.


https://medium.com/mission-log/design-better-data-tables-430a30a00d8c

Best to get a visual designer to weigh in here on usage and prop naming if you want to imply correct usage.
